pg_search/src/postgres/customscan/pdbscan/mod.rs

```diff
@@ -1071,3 +1083,87 @@ pub fn is_block_all_visible(
         status != 0
     }
 }
+
+// Helper function to check if a Bitmapset is empty
+unsafe fn bms_is_empty(bms: *mut pg_sys::Bitmapset) -> bool {
+    let mut set_bit: i32 = -1;
+    // Get the set bits, which are offsets into the RangeTable.
+    set_bit = pg_sys::bms_next_member(bms, set_bit);
+    if set_bit < 0 {
+        return true;
+    }
+    false
+}
+
+// Helper function to convert Bitmapset to Vec<i32>
+unsafe fn bms_to_integers(bms: *mut pg_sys::Bitmapset) -> Vec<i32> {
+    let mut set_bit: i32 = -1;
+    let mut result = Vec::new();
+    loop {
+        // Get the set bits, which are offsets into the RangeTable.
+        set_bit = pg_sys::bms_next_member(bms, set_bit);
+        if set_bit < 0 {
+            break;
+        }
+
+        result.push(set_bit);
+    }
+    result
+}
```

Comment:
The callers of this function would benefit from laziness: would you mind converting it into an `Iterator<Item=i32>`? Could use [`from_fn`](https://doc.rust-lang.org/std/iter/fn.from_fn.html) or a struct.

If you do that, then I think that you can remove `bms_is_empty`.

Reply:
Excellent suggestion. Applied the fix in [3ed0a79](https://github.com/paradedb/paradedb/pull/2440/commits/3ed0a79f5cca44dea837b3cb08809901419cb709)